src/main/java/xyz/nucleoid/codecs/PropagatingOptionalFieldCodec.java

```diff
@@ -6,21 +6,23 @@
 import com.mojang.serialization.MapCodec;
 import com.mojang.serialization.MapLike;
 import com.mojang.serialization.RecordBuilder;
-import net.minecraft.util.dynamic.Codecs;
+import com.mojang.serialization.codecs.OptionalFieldCodec;
 
+import java.util.Objects;
 import java.util.Optional;
 import java.util.stream.Stream;
 
 /**
- * @deprecated Use {@link Codecs.StrictOptionalField}
+ * @deprecated Use {@link OptionalFieldCodec} with {@code lenient} set to {@code false}, which additionally
+ * compares values against the default value using {@link Objects#equals(Object, Object)}.
  */
 @Deprecated
 final class PropagatingOptionalFieldCodec<A> extends MapCodec<Optional<A>> {
```

Comment:
Since we aren't exposing this and it's no longer used internally, can we remove it entirely?